core/src/tile/labels/labels.cpp

```diff
@@ -19,6 +21,8 @@ int Labels::LODDiscardFunc(float _maxZoom, float _zoom) {
 bool Labels::addLabel(MapTile& _tile, const std::string& _styleName, Label::Transform _transform, std::string _text, Label::Type _type) {
     auto currentBuffer = m_ftContext->getCurrentBuffer();
 
+    // FIXME: the current view should not be used to determine whether a label is shown at all
+    // otherwise results will be random
     if ( (m_currentZoom - _tile.getID().z) > LODDiscardFunc(View::s_maxZoom, m_currentZoom)) {
```

Comment:
Right! This should be moved to the update, to discard based on lod interactively.